core/tauri-runtime-wry/src/lib.rs

```diff
@@ -878,6 +878,14 @@ impl WindowBuilder for WindowBuilderWrapper {
     self
   }
 
+  #[cfg(target_os = "macos")]
+  fn transparent_titlebar(mut self, transparent: bool) -> Self {
+    self.inner = self.inner.with_title_hidden(true);
+    self.inner = self.inner.with_fullsize_content_view(true);
+    self.inner = self.inner.with_titlebar_transparent(transparent);
```

Comment:
I think it would be better to expose each method alone, so users can mix and match between them as they see fit.

Reply:
I tried that originally, but the title was hidden and content view fullsize anyway, not sure why. Electron's `titleBarStyle` only really lets you show or hide the whole titlebar, the other options it has (`hiddenInset` and `customButtonsOnHover`) seem to just be for controlling the traffic light buttons

Reply:
> I tried that originally, but the title was hidden and content view fullsize anyway, not sure why. 

Then we need to look into why this happens. I still would like each function to be exposed on its own.

Reply:
Enabling `with_title_hidden ` on it's own does work

Reply:
So the issue is only that `transparent_titlebar` results in `fullsize_content_view` - `hidden_title` works fine, it was just white on a white background for me.

`fullsize_content_view` is being set in the `create_webview` function [here](https://github.com/tauri-apps/tauri/blob/a67005d96bbea36a50532d76e78d3d843b89b16a/core/tauri-runtime-wry/src/lib.rs#L2771-L2774). Any idea why that's being set there, or who would know?

Reply:
tbh I'd prefer reverting that PR since it only happens in devtool, and leave the option for users to enable/disable full content size. Alternatively, since that bug only happened in dev, maybe apply the full content view only in dev mode. cc @lucasfernog

Reply:
exposing all functions separately might be too low level, but i like an interface like `titlebarStyle` from Electron.

Reply:
I'd agree that `fullsize_content_view` is too low level, but I do like having `transparent_titlebar` and `hidden_title` separate

Reply:
we can probably offer both ways :D

Reply:
Something like this?
```
enum TitleBarStyle {
  Visible,
  Transparent,
  Hidden,
}
window.title_bar_style(TitleBarStyle::Transparent);
```